src/lib.rs

```diff
@@ -68,6 +68,7 @@
 //! [`portable-atomic`]: https://crates.io/crates/portable-atomic
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::multiple_bound_locations)] // pin_project_lite can't work around this
```

Comment:
please try to keep separate logical changes in separate commits. It helps with reverting, when needed among other benefits of atomic commits.